sdk/core/azure-core/src/main/java/com/azure/core/util/polling/OperationResourcePollingStrategy.java

```diff
@@ -0,0 +1,265 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util.polling;
+
+import com.azure.core.http.HttpHeader;
+import com.azure.core.http.HttpMethod;
+import com.azure.core.http.HttpPipeline;
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.rest.Response;
+import com.azure.core.implementation.TypeUtil;
+import com.azure.core.util.BinaryData;
+import com.azure.core.util.Context;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.serializer.JacksonAdapter;
+import com.azure.core.util.serializer.SerializerAdapter;
+import com.azure.core.util.serializer.SerializerEncoding;
+import com.azure.core.util.serializer.TypeReference;
+import com.fasterxml.jackson.annotation.JsonSetter;
+import reactor.core.publisher.Mono;
+
+import java.time.Duration;
+
+/**
+ * Implements a operation resource polling strategy, typically from Operation-Location.
+ *
+ * @param <T> the {@link TypeReference} of the response type from a polling call, or BinaryData if raw response body
+ *            should be kept
+ * @param <U> the {@link TypeReference} of the final result object to deserialize into, or BinaryData if raw response
+ *            body should be kept
+ */
+public class OperationResourcePollingStrategy<T, U> implements PollingStrategy<T, U> {
+    private static final String OPERATION_LOCATION = "Operation-Location";
+    private static final String LOCATION = "Location";
+    private static final String REQUEST_URL = "requestURL";
+    private static final String HTTP_METHOD = "httpMethod";
+    private static final String RESOURCE_LOCATION = "resourceLocation";
+    private static final String RETRY_AFTER = "Retry-After";
+    private static final String POLL_RESPONSE_BODY = "pollResponseBody";
+
+    private final SerializerAdapter serializer = new JacksonAdapter();
+    private final ClientLogger logger = new ClientLogger(OperationResourcePollingStrategy.class);
+
+    private final HttpPipeline httpPipeline;
+    private final Context context;
+
+    /**
+     * Creates an instance of the operation resource polling strategy.
+     *
+     * @param httpPipeline an instance of {@link HttpPipeline} to send requests with
+     * @param context additional metadata to pass along with the request
+     */
+    public OperationResourcePollingStrategy(
+            HttpPipeline httpPipeline,
+            Context context) {
+        this.httpPipeline = httpPipeline;
+        this.context = context;
+    }
+
+
+    /**
+     * Gets the name of the operation location header. By default it's "Operation-Location".
+     * @return the name of the operation location header
+     */
+    public String getOperationLocationHeaderName() {
+        return OPERATION_LOCATION;
+    }
```

Comment:
This method is only invoked in this class. It makes sense when subclassing. But `OPERATION_LOCATION` is also read in this class.

Reply:
It was initially designed to be unmodifiable unless subclassed. I've moved this to the constructor to support modifying this field directly.

Reply:
Moved.